src/App.tsx

```diff
@@ -14,11 +14,80 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isVisibleButton: boolean,
+  goodsList: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    isVisibleButton: true,
+    goodsList: goodsFromServer,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      goodsList: [
+        ...state.goodsList.reverse(),
+      ],
+    }));
+  };
+
+  sort = () => {
+    this.setState(state => ({
+      goodsList: [
+        ...state.goodsList.sort(),
+      ],
+    }));
+  };
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goodsList: [
+        ...state.goodsList.sort((a, b) => a.length - b.length),
+      ],
+    }));
+  };
+
+  reset = () => {
+    this.setState(({
+      goodsList: goodsFromServer,
+    }));
+  };
+
+  render() {
+    const { isVisibleButton, goodsList } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Goods:</h1>
+        <button
+          type="button"
+          className={isVisibleButton ? 'visible' : 'hide'}
+          onClick={() => {
+            this.setState({
+              isVisibleButton: false,
+            });
+          }}
+        >
+          start
+        </button>
+        <button type="button" onClick={this.reverse}>Reverse</button>
```

Comment:
move to new lines, this way it's more readable

```suggestion
        <button
          type="button"
          onClick={this.reverse}
        >
          Reverse
        </button>
```